src/Model/UserAction.php

```diff
@@ -0,0 +1,228 @@
+<?php
+
+declare(strict_types=1);
+
+namespace atk4\data\Model;
+
+use atk4\core\DIContainerTrait;
+use atk4\core\Exception;
+use atk4\core\InitializerTrait;
+use atk4\core\TrackableTrait;
+use atk4\data\Model;
+
+/**
+ * Implements generic user action. Assigned to a model it can be invoked by a user. Model\UserAction class contains a
+ * meta information about the action (arguments, permissions, scope, etc) that will help UI/API or add-ons to display
+ * action trigger (button) correctly in an automated way.
+ *
+ * UserAction must NOT rely on any specific UI implementation.
+ *
+ * @property Model $owner
+ */
+class UserAction
+{
+    use DIContainerTrait;
+    use TrackableTrait;
+    use InitializerTrait {
+        init as init_;
+    }
+
+    /** Defining scope of the action */
+    const SCOPE_NONE = 'none'; // e.g. add
```

Comment:
@georgehristov please explain one more time how it is related to `Scope`?

For me, it is very confusing and like with field/control terminology I think we should not use `scope` here as it refers to "when action can be applied" instead of "records filtering"

Can we change this constant to `APPLIES_TO_XXX_RECORD(S)`?

Reply:
`AFFECTS_XXX_RECORD(S)`?

Reply:
affects means it will change data, but I think it does not have to...

Reply:
Yes, after a lot of thinking I changed now to APPLIES_TO prefix